api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -264,4 +295,498 @@ public void saveConcept_shouldForceSetFlagIfSetMembersExist() throws Exception {
+		assertNull(conceptService.getConcept(conceptId));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#retireConcept(Concept,String)
+	 * @Verifies should fail if no reason is given
+	 */
+	@Test
+	public void retireConcept_shouldFailIfNoReasonIsGiven() throws Exception {
+		int conceptId = 3;
+		Concept concept = conceptService.getConcept(conceptId);
+		expectedException.expect(IllegalArgumentException.class);
+		conceptService.retireConcept(concept, "");
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#retireConcept(Concept,String)
+	 * @Verifies should retire the given concept
+	 */
+	@Test
+	public void retireConcept_shouldRetireTheGivenConcept() throws Exception {
+		int drugId = 3;
+		String retireReason = "dummy reason";
+		Concept concept = conceptService.getConcept(drugId);
+		assertFalse(concept.isRetired());
+		conceptService.retireConcept(concept, retireReason);
+		assertTrue(concept.isRetired());
+		assertEquals(retireReason, concept.getRetireReason());
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#purgeDrug(Drug)
+	 * @Verifies should purge the given Drug
+	 */
+	@Test
+	public void purgeDrug_shouldPurgeTheGivenDrug() throws Exception {
+		int drugId = 2;
+		conceptService.purgeDrug(conceptService.getDrug(drugId));
+		assertNull(conceptService.getDrug(drugId));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getAllDrugs()
+	 * @Verifies should return a list of all drugs
+	 */
+	@Test
+	public void getAllDrugs_shouldReturnAListOfAllDrugs() throws Exception {
+		int resultWhenTrue = 4;
+		int resultWhenFalse = 2;
+		List<Drug> allDrugs = conceptService.getAllDrugs();
+		assertEquals(resultWhenTrue, allDrugs.size());
+		allDrugs = conceptService.getAllDrugs(false);
+		assertEquals(resultWhenFalse, allDrugs.size());
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getDrugs(String, Concept, boolean, boolean, boolean, Integer,
+	 *      Integer)
+	 * @Verifies return list of drugs
+	 */
+	@Test
+	public void getDrugs_shouldReturnListOfMatchingDrugs() throws Exception {
+		String drugName = "ASPIRIN";
+		String drugUuid = "05ec820a-d297-44e3-be6e-698531d9dd3f";
+		Concept concept = conceptService.getConceptByUuid(drugUuid);
+		List<Drug> drugs = conceptService.getDrugs(drugName, concept, true, true, true, 0, 100);
+		assertTrue(drugs.contains(conceptService.getDrug(drugName)));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getDrug(String)
+	 * @Verifies should return the matching drug object
+	 */
+	@Test
+	public void getDrug_shouldReturnTheMatchingDrugObject() throws Exception {
+		String drugName = "ASPIRIN";
+		String drugUuid = "05ec820a-d297-44e3-be6e-698531d9dd3f";
+		Drug drug = conceptService.getDrugByUuid(drugUuid);
+		assertEquals(drug, conceptService.getDrug(drugName));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getDrug(String)
+	 * @Verifies should return null if no matching drug is found
+	 */
+	@Test
+	public void getDrug_shouldReturnNullIfNoMatchingDrugIsFound() throws Exception {
+		int drugIdNotPresent = 1234;
```

Comment:
thats nice, when you want to explain a certain value you can use this style. explanatory temp variable. instead of just a magic number.